sql/hive/src/test/scala/org/apache/spark/sql/hive/CompressionCodecSuite.scala

```diff
@@ -0,0 +1,321 @@
+        assert(relCompressionCodecs.length == 1)
+        val tableSize = getTableSize(path)
+        assertion(relCompressionCodecs.head, tableSize)
+      }
+    }
+  }
+
+  private def checkTableCompressionCodecForCodecs(
+      format: String,
+      isPartitioned: Boolean,
+      convertMetastore: Boolean,
+      compressionCodecs: List[String],
+      tableCompressionCodecs: List[String])
+      (assertionCompressionCodec: (Option[String], String, String, Long) => Unit): Unit = {
+    withSQLConf(getConvertMetastoreConfName(format) -> convertMetastore.toString) {
+      tableCompressionCodecs.foreach { tableCompression =>
+        compressionCodecs.foreach { sessionCompressionCodec =>
+          withSQLConf(getSparkCompressionConfName(format) -> sessionCompressionCodec) {
+            // 'tableCompression = null' means no table-level compression
+            val compression = Option(tableCompression)
+            checkCompressionCodecForTable(format, isPartitioned, compression) {
+              case (realCompressionCodec, tableSize) => assertionCompressionCodec(compression,
+                sessionCompressionCodec, realCompressionCodec, tableSize)
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // When the amount of data is small, compressed data size may be larger than uncompressed one,
+  // so we just check the difference when compressionCodec is not NONE or UNCOMPRESSED.
+  private def checkTableSize(
+      format: String,
+      compressionCodec: String,
+      isPartitioned: Boolean,
+      convertMetastore: Boolean,
+      tableSize: Long): Boolean = {
+    format match {
+      case "parquet" =>
+        val uncompressedSize = if (!convertMetastore || isPartitioned) {
+          getUncompressedDataSizeByFormat(format, isPartitioned = true)
+        } else {
+          getUncompressedDataSizeByFormat(format, isPartitioned = false)
+        }
+
+        if (compressionCodec == "UNCOMPRESSED") {
+          tableSize == uncompressedSize
+        } else {
+          tableSize != uncompressedSize
+        }
+      case "orc" =>
+        val uncompressedSize = if (!convertMetastore || isPartitioned) {
+          getUncompressedDataSizeByFormat(format, isPartitioned = true)
+        } else {
+          getUncompressedDataSizeByFormat(format, isPartitioned = false)
+        }
+
+        if (compressionCodec == "NONE") {
+          tableSize == uncompressedSize
+        } else {
+          tableSize != uncompressedSize
+        }
+      case _ => false
+    }
+  }
+
+  def checkForTableWithCompressProp(format: String, compressCodecs: List[String]): Unit = {
+    Seq(true, false).foreach { isPartitioned =>
+      Seq(true, false).foreach { convertMetastore =>
+        checkTableCompressionCodecForCodecs(
+          format,
+          isPartitioned,
+          convertMetastore,
+          compressionCodecs = compressCodecs,
+          tableCompressionCodecs = compressCodecs) {
```

Comment:
Also add another scenario. 
```
compressionCodecs = Nil,
tableCompressionCodecs = compressCodecs 
```

Reply:
Do you mean only with table-level compression? Actually, `compressionCodecs` is the session-level compressionCodec, even if set to `Nil` or `null` here, it still takes the default value `snappy`.  
In the first test case ``` test("both table-level and session-level compression are set") ```, isn't it already contained?

Reply:
OK.